src/video_stream/types.rs

```diff
@@ -38,7 +38,7 @@ impl VideoAndStreamInformation {
         let other_endpoints: HashSet<_> = other.stream_information.endpoints.iter().collect();
         let common_endpoints: HashSet<_> = our_endpoints.intersection(&other_endpoints).collect();
 
-        if !common_endpoints.is_empty() {
+        if !common_endpoints.is_empty() && self.stream_information.endpoints[0].scheme() != "turn" {
```

Comment:
We should add a `allow_same_endpoint` bool for webrtc stream type. and avoid this check here.